src/server/worker/master.go

```diff
@@ -72,7 +74,7 @@ func (logger *taggedLogger) jobLogger(jobID string) *taggedLogger {
 	return result
 }
 
-func (a *APIServer) master() {
+func (a *APIServer) master(spawnerName string, spawner func(*client.APIClient) error) {
```

Comment:
Passing in a string here is a great way to solve the problem. Minor nits:
- could we call the variable `masterType` that seems to match up with how it's use a bit better.
- for the values passed in could we have them be: `"service"`, `"spout"` and `"pipeline"`, I think that will lead to the most understandable logs for users.